airbyte_cdk/sources/declarative/retrievers/simple_retriever.py

```diff
@@ -626,6 +627,102 @@ def _to_partition_key(to_serialize: Any) -> str:
         # separators have changed in Python 3.4. To avoid being impacted by further change, we explicitly specify our own value
         return json.dumps(to_serialize, indent=None, separators=(",", ":"), sort_keys=True)
 
+    def _fetch_one(
```

Comment:
Given that this is presumably going to be called from somewhere else outside of `SimpleRetriever`, we should make this public. While I understand this is not meant to be public facing or used by the protocol interface, it still feels like bad practice for this be a private method.